src/python/test/test_remote_datasets.py

```diff
@@ -0,0 +1,119 @@
+    # Test fetch with a wrong checksum
+    with pytest.raises(OSError):
+        _check_fetch_output("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/spiral_2d/spiral_2d.npy", "spiral_2d.npy", file_checksum = 'XXXXXXXXXX')
+
+    # Test files download from given urls with checksums provided
+    _check_fetch_output("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/spiral_2d/spiral_2d.npy", "spiral_2d.npy",
+                                file_checksum = '88312ffd6df2e2cb2bde9c0e1f962d7d644c6f58dc369c7b377b298dacdc4eaf')
+
+    _check_fetch_output("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/sphere3D_pts_on_grid.off", "sphere3D_pts_on_grid.off",
+                                file_checksum = '32f96d2cafb1177f0dd5e0a019b6ff5658e14a619a7815ae55ad0fc5e8bd3f88')
+
+    # Test files download from given urls without checksums
+    _check_fetch_output("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/spiral_2d/spiral_2d.npy", "spiral_2d.npy")
+
+    _check_fetch_output("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/sphere3D_pts_on_grid.off", "sphere3D_pts_on_grid.off")
+
+    # Test printing existing LICENSE file when fetching bunny.npy with accept_license = False (default)
+    # Fetch LICENSE file
+    makedirs("remote_datasets/bunny", exist_ok=True)
+    remote._fetch_remote("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/bunny/LICENSE", "LICENSE", "remote_datasets/bunny",
+                 'b763dbe1b2fc6015d05cbf7bcc686412a2eb100a1f2220296e3b4a644c69633a')
+    with open("remote_datasets/bunny/LICENSE") as f:
+        assert f.read().rstrip("\n") == _get_bunny_license_print().getvalue().rstrip("\n")
+
+    # Test not printing bunny.npy LICENSE when accept_license = True
+    assert "" == _get_bunny_license_print(accept_license = True).getvalue()
+
+    # Remove "remote_datasets" directory and all its content
+    shutil.rmtree("remote_datasets")
+
+def test_fetch_remote_datasets_wrapped():
+    # Test fetch_spiral_2d and fetch_bunny wrapping functions (twice, to test case of already fetched files)
+    for i in range(2):
+        spiral_2d_arr = remote.fetch_spiral_2d()
+        assert spiral_2d_arr.shape == (114562, 2)
+
+        bunny_arr = remote.fetch_bunny()
+        assert bunny_arr.shape == (35947, 3)
+
+    # Check that default dir was created
+    assert isdir(expanduser("~/remote_datasets"))
+
+    # Test fetch_spiral_2d and fetch_bunny wrapping functions with data directory different from default
+    spiral_2d_arr = remote.fetch_spiral_2d(dirname = "~/another_fetch_folder")
+    assert spiral_2d_arr.shape == (114562, 2)
+
+    bunny_arr = remote.fetch_bunny(dirname = "~/another_fetch_folder")
+    assert bunny_arr.shape == (35947, 3)
+
+    assert isdir(expanduser("~/another_fetch_folder"))
+
+    # Remove test folders
+    del spiral_2d_arr
+    del bunny_arr
+    shutil.rmtree(expanduser("~/remote_datasets"))
```

Comment:
I would not remove all the user default gudhi data folder. Imagine if we have a dataset of 2 Go, launching the unitary test would remove it. Maybe just delete spiral and bunny (these datasets are quite small).

Reply:
Testing remove on the non-default cache directory should be enough.

Reply:
Removing the default gudhi data folder was for cleaning reasons. If I don't have one, I wouldn't want it to be created in my home directory just because I ran some tests. Now, it is removed only if it didn't exist beforehand.